Run specs against real redis as well; refactor Action's scope

kigster opened the pull request:
We are using Pause on a project, and I was getting an error that seemed like a redis error. Without being able to run specs against a real redis server, it was impossible to tell if the problem was with the gem or somewhere else.

Now the entire spec suite can run using `fakeredis` (the default), and against a local redis instance. 

To access integration tests, run `rake spec:integration` instead of `rake spec`.

Additionally, refactored the `Pause::Action` to stop using a class variable, which are [considered harmful](http://archive.oreilly.com/pub/a/ruby/excerpts/ruby-best-practices/worst-practices.html).

Updated Travis config to run against ruby 2.4.0, and to run both test suites.

Commit:
Updating Ruby test matrix

Commit:
Use gem spec as required by the specification

Commit:
Simplify method overriding for ShardedAdapter

Commit:
Kill fakeredis as it’s masking real issues.

Commit:
Add redis-server to Travis

Commit:
Fix rspecs to work with real redis

 * refactored Action class-level methods
 * got rid of a horrendous class level variable
 * derive default action scope from the class name
 * flush redis db before each test.
 * clean up gemspec
 * bump version to 0.3.0

Commit:
Enabling both fakeredis and real redis tests

Commit:
Add a message about redis-server

wanelo-pair commented:
@kigster : Travis fails on 1.9.3 

``
Gem::InstallError: ruby_dep requires Ruby version >= 2.2.5, ~> 2.2.
``

Thoughts on supporting anything less than 2.2?

kigster commented:
I am happy to drop 1.9.3 if you guys are :)

Commit:
Remove 1.9.3

Commit:
More defensive zrem() command; more tests.

The issue I faced is a race condition when two processes increment at
the same time; first one performs zrem on two old elements, second
does not have any elements to zrem (even though zcard > value) was
satisified. This race condition must be simply coded around.

hjhart commented:
Looks good to @mattcamuto and I.

@kigster Should we try to do some real redis tests in travis?

kigster commented:
We do on this branch! Check out `.travis.yml`; we start redis server, and the test suite runs twice: once with redis, and once without.

kigster committed:
Introducing block helpers to simplify the API.

- together with tests
- updating gems, removing unused gems
- migrating gem to /kigster for maintenance
- bump version to 0.4.0

kigster committed:
Making Travis more robust

kigster committed:
Adding SimpleCov text coverage (currently at 99.13%)

kigster committed:
Block usage for Action constructor

 - also adding CodeCimate, new tests
 - updating README
 - version bump 0.4.1

kigster committed:
Add badges to README

kigster committed:
Make builds more robust
